### PR TITLE
faction objective system

### PR DESCRIPTION
**Pull request by lpeapnni**
## About The Pull Request
adds objective system for factions

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
add: Added faction objective system
/:cl:

### CODE REVIEW

File: code/modules/antagonists/faction/faction.dm [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		to_chat(faction_member, "<span class='notice'>A new objective has been assigned: [additional_objective]</span>")
+
+// here are the actual factions
+
+/datum/team/faction/legion
+	faction_id = "LEG"
+
+/datum/antagonist/faction/legion
+	show_in_antagpanel = TRUE
+	name = "Legionarie"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	name = "Legionary"
```

File: code/modules/antagonists/faction/faction.dm [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	roundend_category = "Ceasar's Legion"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	roundend_category = "The Boreal Legion"
```

File: code/modules/antagonists/faction/faction.dm [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	faction_name = "Ceasar's Legion"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	faction_name = "The Boreal Legion"
```

File: code/modules/antagonists/faction/faction.dm [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	faction_id = "LEG"
+	faction_team_type = /datum/team/faction/legion
+
+/datum/team/faction/ncr
+	faction_id = "NCR"
+
+/datum/antagonist/faction/ncr
+	show_in_antagpanel = TRUE
+	name = "NCR Soldier"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	name = "PRNC Soldier"
```

File: code/modules/antagonists/faction/faction.dm [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+	roundend_category = "The New California Republic"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	roundend_category = "The People's Republic of New California"
```

File: code/modules/antagonists/faction/faction.dm [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	faction_name = "New California Republic Army"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	faction_name = "The Californian People's Liberation Force"
```

File: code/modules/antagonists/faction/faction.dm [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+	faction_id = "NCR"
+	faction_team_type = /datum/team/faction/ncr
+
+/datum/team/faction/bos
+	faction_id = "BOS"
+
+/datum/antagonist/faction/bos
+	show_in_antagpanel = TRUE
+	name = "BOS Member"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	name = "Brotherhood of Steel Cultist"
```

File: code/modules/antagonists/faction/faction.dm [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+/datum/antagonist/faction/tribal
+	show_in_antagpanel = TRUE
+	name = "Tribal"
+	roundend_category = "The tribals"
+	faction_name = "Tribals"
+	faction_id = "TRIB"
+	faction_team_type = /datum/team/faction/tribal
+
+/datum/team/faction/bighorn
+	faction_id = "BH"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	faction_id = "TOWN"
```

File: code/modules/antagonists/faction/faction.dm [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+/datum/antagonist/faction/bighorn
+	show_in_antagpanel = TRUE
+	name = "Bighorn Citizen"
+	roundend_category = "Bighorn"
+	faction_name = "Bighorn"
+	faction_id = "BH"
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/faction.dm]:
```suggestion
	faction_id = "TOWN"
```

File: code/modules/antagonists/faction/add_faction_objective.dm [hunk 10]
```diff
@@ -0,0 +1,49 @@
+/mob/living/proc/add_legion_objective()
+	set name = "Add Legion Objective"
+	set category = "IC"
+	add_faction_objective(/datum/antagonist/faction/legion, list("Legion Legate", "Legion Centurion", "Legion Veteran Decanus", "Legion Prime Decanus", "Legion Recruit Decanus"))
```
Comment (Draganfrukts) [inline code/modules/antagonists/faction/add_faction_objective.dm]:
```suggestion
	add_faction_objective(/datum/antagonist/faction/legion, list("Legion Legate", "Legion Centurion", "Legion Veteran Decanus"))
```

**Review by Draganfrukts — CHANGES_REQUESTED** [review]
Fine otherwise, remember to add these 
Change the Bighorn stuff into just generic town for now

File: code/modules/antagonists/faction/faction.dm (hunk 9 above)
Comment (lpeapnni) [reply code/modules/antagonists/faction/faction.dm]: FUCK YOU!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! FUCK YOU!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by lpeapnni** [comment]
im just gonna make this not a draft anymore because we can pr non placeholder objectives later

**Review by Draganfrukts — APPROVED** [review]
good

**Pull request merged**